### PR TITLE
macro hygiene usage fixes

### LINKED ISSUE #23146: some deprecated methods in the wrong module
E.g. the vectorized method of `acos_fast` is in Base instead of FastMath. Caused by 7ec9b1857e029d4307fa0968512f828481611440. cc @vtjnash 

I believe this is because in

```
macro dep_vectorize_1arg(S, f)
    x = esc(:x) # work around macro hygiene bug
    T = esc(:T) # work around macro hygiene bug
    return :( @deprecate $f($x::AbstractArray{$T}) where {$T<:$S} $f.($x) )
end
```

since `f` refers to a raw symbol, the `$f` is now looked up in the definition environment of the macro, which is Base. I tried to fix it by adding back `f = esc(f)`, but the `@deprecate` macro doesn't like that. I usually end up with:

```
LoadError("sysimg.jl", 416, LoadError("deprecated.jl", 263, ErrorException("syntax: malformed \"export\" statement")))
```
or
```
LoadError("sysimg.jl", 416, LoadError("deprecated.jl", 151, ErrorException("syntax: invalid syntax (escape (outerref isambiguous))")))
```

I suspect both are due to not allowing multiple levels of `esc` wrapping.

### PR DESCRIPTION
**Pull request by vtjnash**
fixes #23146

### CODE REVIEW

**Comment by JeffBezanson** [comment]
Thanks.

Here's another one. In this case in test/parse.jl:

```
module M16096
macro iter()
    return quote
        @inline function foo16096(sub)
            it = 1
        end
    end
end
end
```

This macro creates a global function `M16096.foo16096`. However, if `@inline` is removed then it creates a renamed local function at the call site (which is what it did in 0.6, and which I believe is correct).

**Comment by vtjnash** [comment]
That's weird. But it's unrelated. Here's a MRE of it:
```julia
julia> macroexpand(Main, Expr(Symbol("hygienic-scope"), :(begin; function a() end; end), M16096))
quote
    #= REPL[37]:1 =#
    function #12#a()
        #= REPL[37]:1 =#
    end
end

julia> macroexpand(Main, Expr(Symbol("hygienic-scope"), :(function a() end), M16096))
:(function (M16096.a)()
        #= REPL[35]:1 =#
    end)
```
I don't think we can ever be free of this sort of error until this code is deleted. Implementing two copies of resolve-scopes is just too error-prone.

**Comment by vtjnash** [comment]
(Oh, and fwiw, the v0.6 behavior was wrong, but since it was due to the culmination of several bugs, I think I would just characterize its behavior here as random bad luck. Now at least we get 50% of the cases correct :P)

**Comment by JeffBezanson** [comment]
Yeah that's fine. It would be great to delete all that code.

File: base/deprecated.jl [hunk 1]
```diff
@@ -28,8 +28,7 @@ macro deprecate(old, new, ex=true)
             ex ? Expr(:export, esc(old)) : nothing,
             :(function $(esc(old))(args...)
                   $meta
-                  depwarn(string($oldname, " is deprecated, use ", $newname, " instead."),
-                          $oldmtname)
+                  depwarn($"$old is deprecated, use $new instead.", $oldmtname)
```
Comment (pabloferz) [inline base/deprecated.jl]: If you ise `old` and `new` directly, you can remove the `oldname` and `newname` expressions above, right?

**Pull request merged**